### PR TITLE
Support Compression in parquet-fromcsv

### LINKED ISSUE #3721: Support compressed CSV/TSV files in `parquet-fromcsv`.
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Reading a gzip compressed TSV file with `parquet-fromcsv`.

**Describe the solution you'd like**

Support compressed CSV/TSV files in `parquet-fromcsv`.

Also it would be nice if there was a link to the parquet schema text format, or some example schemas.

### PR DESCRIPTION
**Pull request by suxiaogang223**
# Which issue does this PR close?

Closes #3721 

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?
add new arg for parquet-fromcsv : -C/--csv-commoression

### CODE REVIEW

File: parquet/src/bin/parquet-fromcsv.rs [hunk 1]
```diff
@@ -626,14 +658,71 @@ mod tests {
         schema.as_file().write_all(schema_text.as_bytes()).unwrap();
 
         let mut input_file = NamedTempFile::new().unwrap();
-        {
-            let csv = input_file.as_file_mut();
+
+        fn wirte_tmp_file<T: Write>(w: &mut T) {
```
Comment (tustvold) [inline parquet/src/bin/parquet-fromcsv.rs:662]:
```suggestion
        fn write_tmp_file<T: Write>(w: &mut T) {
```

File: parquet/src/bin/parquet-fromcsv.rs [hunk 2]
```diff
@@ -368,9 +376,28 @@ fn convert_csv_to_parquet(args: &Args) -> Result<(), ParquetFromCsvError> {
...
+        Compression::BROTLI(_) => {
+            Box::new(Decompressor::new(input_file, 0)) as Box<dyn Read>
+        }
+        Compression::LZ4 => Box::new(lz4::Decoder::new(input_file).map_err(|e| {
+            ParquetFromCsvError::with_context(e, "Failed to create lz4::Decoder")
+        })?) as Box<dyn Read>,
+        Compression::ZSTD(_) => Box::new(zstd::Decoder::new(input_file).map_err(|e| {
+            ParquetFromCsvError::with_context(e, "Failed to create zstd::Decoder")
+        })?) as Box<dyn Read>,
+        // TODO: I wonder which crates should i use to decompress lzo and lz4_raw?
```
Comment (tustvold) [inline parquet/src/bin/parquet-fromcsv.rs:394]: They are codecs that only make sense in the context of parquet's block compression

**Review by tustvold — COMMENTED** [review]
Looking good thank you, only thing left I think is some feature flag shenanigans

**Comment by suxiaogang223** [comment]
> Looking good thank you, only thing left I think is some feature flag shenanigans
> 
> I believe running with `cargo run --bin parquet-from-csv --no-default-features --features arrow,cli` will result in compilation errors now

thanks, I'll fix this soon

**Review by tustvold — APPROVED** [review]

**Pull request merged**